launcher/gui.py

```diff
@@ -0,0 +1,322 @@
+        self.thread = threading.Thread()
+        self.output = StringIO()
+        self.textbuffer = Gtk.TextBuffer()
+        self.stderr = sys.stderr
+        self.stdout = sys.stdout
+        if len(sys.argv) > 1 and sys.argv[1] == '--debug':
+            print("stdout and stderr aren't redirected")
+        else:
+            print("Redirecting stdout and stderr")
+            sys.stdout = self.output
+            sys.stderr = self.output
+        self.win = StackWindow(application=app)
+        self.mods = []
+        self.entry_buffers = {}
+        self.load_button = None
+
+        self.win.init_stack()
+        self.win.present()
+
+        self.generic_tab('Install Anomaly', ['Anomaly Dir', 'Cache Dir'], self.anomaly_install)
+        self.generic_tab('Full Gamma Setup', ['Anomaly Dir', 'Gamma Dir', 'Custom Gamma Definition', 'Custom Gamma Repository', 'Cache Dir'], self.full_install)
+        self.generic_tab('Partial Gamma Setup', ['Anomaly Dir', 'Gamma Dir', 'Custom Gamma Definition', 'Custom Gamma Repository', 'Cache Dir'], self.gamma_setup)
+        self.generic_tab('Full Gamma Install', ['Anomaly Dir', 'Gamma Dir', 'Final Gamma Dir'], self.gamma_usvfs)
+        self.mod_chooser_tab()
+
+    def generic_tab(self, name, list_inputs, install_fnct):
+        box_outer = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=24)
+        listbox = Gtk.ListBox()
+        listbox.props.selection_mode = Gtk.SelectionMode.NONE
+        listbox.props.show_separators = True
+        row = Gtk.ListBoxRow()
+        hbox = Gtk.ListBox()
+        row.set_child(hbox)  # We set the Box as the ListBoxRow child
+        self.entries[name] = []
+        for i in list_inputs:
+            hbox.append(Gtk.Label(label=i))
+            if i not in self.entry_buffers:
+                self.entry_buffers[i] = Gtk.EntryBuffer()
+            entry = Gtk.Entry.new_with_buffer(self.entry_buffers[i])
+            self.entries[name].append(entry)
+            hbox.append(entry)
+        self.button = Gtk.Button()
+        self.button.set_label('Execute')
+        self.button.connect('clicked', install_fnct)
+        hbox.append(self.button)
+        # ===============
+        textview = Gtk.TextView.new_with_buffer(self.textbuffer)
+        textview.set_editable(False)
+        textview.set_cursor_visible(False)
+        scroll = Gtk.ScrolledWindow()
+        scroll.set_child(textview)
+        scroll.set_propagate_natural_height(True)
+        exp = Gtk.Expander()
+        exp.set_child(scroll)
+        exp.set_expanded(True)
+        exp.set_resize_toplevel(True)
+        hbox.append(exp)
+        # =============
+        listbox.append(row)
+        box_outer.append(listbox)
+        self.win.add_titled_to_stack(box_outer, name, name)
+        self.win.init_stack()
+
+    def load_mods(self, button):
+        self.file_name = self.entries['Mod Chooser'][0].get_chars(0, -1)
+        self.file_name += '/profiles/G.A.M.M.A/modlist.txt'
```

Comment:
Use `pathlib.Path` to avoid / issue ... also, it will work on Windows too.

`self.file_name = Path(self.file_name) / 'profile' / 'G.A.M.M.A' / 'modlist.txt'`